omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -125,13 +128,19 @@ public int compare(SimpleObject o1, SimpleObject o2) {
 	}
 	
 	private SimpleObject parseObs(Obs obs, Locale locale) {
+        Concept concept = obs.getConcept();
+        if (concept != null && concept.isNumeric()) {
+                //TODO HibernateUtil.getRealObjectFromProxy(obs.getConcept()) should be moved to a better place in the future,
+                //If we remove it, the concept type will never be an instanceof ConceptNumeric
+                obs.setConcept(HibernateUtil.getRealObjectFromProxy(concept));
+        }
         if ("org.openmrs.Location".equals(obs.getComment())) {
             return (parseObsWithLocationAnswer(obs, locationService.getLocation(Integer.valueOf(obs.getValueText()))));
         }
         else {
             SimpleObject simpleObject = SimpleObject.create("obsId", obs.getObsId());
 
-            simpleObject.put("question", capitalizeString(uiUtils.format(obs.getConcept())));
+            simpleObject.put("question", capitalizeString(uiUtils.format(concept)));
```

Comment:
Isn't `concept` still a hibernate proxy?

Reply:
Yes, sorry, it can be `obs.getConcept()` to use the value with `HibernateUtil`